zebrad/src/commands/start/sync.rs

```diff
@@ -107,12 +178,20 @@ where
             // Wipe state from prevous iterations.
             self.prospective_tips = HashSet::new();
             self.pending_blocks = Box::pin(FuturesUnordered::new());
+            self.update_metrics();
 
             tracing::info!("starting sync, obtaining new tips");
-            if self.obtain_tips().await.is_err() {
-                tracing::warn!("failed to obtain tips, waiting to restart sync");
-                delay_for(SYNC_RESTART_TIMEOUT).await;
-                continue 'sync;
+            if self.obtain_tips().await.is_err() || self.prospective_tips.is_empty() {
+                // Retry ObtainTips once
+                tracing::info!("failed to obtain tips, waiting to retry obtain tips");
+                delay_for(TIPS_RETRY_TIMEOUT).await;
+                let _ = self.obtain_tips().await;
+
+                if self.prospective_tips.is_empty() {
+                    tracing::warn!("failed to obtain tips, waiting to restart sync");
+                    delay_for(SYNC_RESTART_TIMEOUT).await;
+                    continue 'sync;
+                }
```

Comment:
Non-blocking on this PR, but when we restart the sync process, we retry `obtain_tips`.  Is there a benefit to special-casing this retry logic here?

Reply:
I don't think there is much of a benefit in the current code.

The TIPS_RETRY_TIMEOUT is shorter, but not by much. So always using the SYNC_RESTART_TIMEOUT would be simpler and more reliable.

I think it's a holdover from when the gap between the TIPS_RETRY_TIMEOUT and SYNC_RESTART_TIMEOUT was much larger.

Reply:
See #1033 for a fix for this issue.